includes/GuideObjects/CoordinateSystem.js

```diff
@@ -1,36 +1,159 @@
+	ASSERT(uw_plane_point_coordinate.length === 3, "Second point (uw plane) must be in form [X, Y, Z] defined");
+	this.coordinate_system = createBaseCoordinateSystem(no, comment, params);
+	this.coordinate_system.type = coordinate_systems.TYPE_3_POINTS;
+	this.coordinate_system.origin_coordinate_x = original_point[0];
+    this.coordinate_system.origin_coordinate_y = original_point[1];
+    this.coordinate_system.origin_coordinate_z = original_point[2];
+	this.coordinate_system.u_axis_point_coordinate_x = u_axis_point_coordinate[0];
+	this.coordinate_system.u_axis_point_coordinate_y = u_axis_point_coordinate[1];
+	this.coordinate_system.u_axis_point_coordinate_z = u_axis_point_coordinate[2];
+	this.coordinate_system.uw_plane_point_coordinate_x = uw_plane_point_coordinate[0];
+	this.coordinate_system.uw_plane_point_coordinate_y = uw_plane_point_coordinate[1];
+	this.coordinate_system.uw_plane_point_coordinate_z = uw_plane_point_coordinate[2];
+	return this.coordinate_system;
+};
 
-    set_coordinates('origin_coordinates', ['origin_coordinate_x', 'origin_coordinate_y', 'origin_coordinate_z']);
-    set_coordinates('u_axis_point_coordinates', ['u_axis_point_coordinate_x', 'u_axis_point_coordinate_y', 'u_axis_point_coordinate_z']);
-    set_coordinates('uw_plane_point_coordinates', ['uw_plane_point_coordinate_x', 'uw_plane_point_coordinate_y', 'uw_plane_point_coordinate_z']);
+/**
+ * Create  coordinate system with "2 Points and Angle" type
+ * @param {Number}	no 							Index of coordinate system, can be undefined
+ * @param {Array} 	original_point 				Coordinates of original point
+ * @param {Array}	uw_plane_point_coordinate	Coordinates of uw plane point
+ * @param {Array}	uw_plane_angle				Rotation of UW-plane
+ * @param {String} 	comment 					Comment, can be undefined
+ * @param {Object} 	params 						Additional parameters, can be undefined
+ * @returns 									Created coordinate system
+ */
+CoordinateSystem.prototype.TwoPointsAndAngle = function (no,
+	original_point,
+	u_axis_point_coordinate,
+	uw_plane_angle,
+	comment,
+	params) {
+	ASSERT(original_point.length === 3, "Original point has must be in form [X, Y, Z] defined");
+	ASSERT(u_axis_point_coordinate.length === 3, "U axis point must be in form [X, Y, Z] defined");
+	this.coordinate_system = createBaseCoordinateSystem(no, comment, params);
+	this.coordinate_system.type = coordinate_systems.TYPE_2_POINTS_AND_ANGLE;
+	this.coordinate_system.origin_coordinate_x = original_point[0];
+    this.coordinate_system.origin_coordinate_y = original_point[1];
+    this.coordinate_system.origin_coordinate_z = original_point[2];
+	this.coordinate_system.u_axis_point_coordinate_x = u_axis_point_coordinate[0];
+	this.coordinate_system.u_axis_point_coordinate_y = u_axis_point_coordinate[1];
+	this.coordinate_system.u_axis_point_coordinate_z = u_axis_point_coordinate[2];
+	if (uw_plane_angle !== "undefined") {
+		this.coordinate_system.uw_plane_angle = uw_plane_angle;
```

Comment:
Add function to Tools - recalculation of angle from degrees to radians (and from radians to degrees)